strats.md

```diff
@@ -163,6 +165,62 @@ The `leaveWithSpark` object currently has no properties. If needed, properties m
+
+A `leaveWithGModeSetup` object contains the following property:
+- _knockback_: If true, then Samus gets knockback frames through the transition. This makes it possible for Samus to retain mobility in the next room if reserve energy is at most 4. Most enemies provide knockback, so this property is true by default if unspecified. Certain enemies such as Beetoms, Metroids, and Mochtroids do not provide knockback, so this property should be set to false for strats that use these enemies for the transition damage.
+
+A `leaveWithGModeSetup` comes with implicit requirements, which are described in detail under the entrance conditions `comeInWithRMode` and `comeInWithGMode`. The implicit requirements depend on the specifics of the entrance condition in the next room, but they always include at least the following:
+- The `XRayScope` item requirement.
+- A requirement to have at least 1 reserve energy.
+- A requirement to damage down to 0 energy, triggering reserves.
+
+#### Example
+```json
+{
+  "name": "Leave With G-Mode Setup",
+  "notable": false,
+  "requires": [],
+  "exitCondition": {
+    "leaveWithGModeSetup": {}
+  }
+}
+```
+
+### Leave with G-Mode
+
+A `leaveWithGMode` exit condition represents that Samus can leave through this door while in G-mode, resulting in indirect G-mode in the next room. A strat with a `leaveWithGMode` exit condition should always also have a `comeInWithGMode` entrance condition since the only (known) way to enter G-mode is while (or before) coming into the room.
+
+A `leaveWithGMode` object has the following property:
+- _morphed_: If true, then this strat results in leaving the room in a morphed state, either by maintaining artificial morph or by having the Morph item.
+
+For most doors in the game, it is possible to enter the room with a G-mode setup and then immediately exit back through the same door. This is because in direct G-mode the door does not close behind Samus. To avoid the need to write out tedious boilerplate, these strats are understood to be included implicitly. Specifically, for every door node without a `spawnAt` property, there are two implicit strats, with `leaveWithGMode` exit conditions, one of the form
+
+```json
+{
+  "name": "G-Mode Go Back Through Door",
+  "notable": false,
+  "entranceCondition": {
+    "comeInWithGMode": {
+      "mode": "direct",
+      "morphed": false
+    }
+  },
+  "requires": [],
+  "exitCondition": {
+    "leaveWithGMode": {
+      "morphed": false
+    }
+  }
+}
+```
+
+and another where the `false` values for `morphed` are replaced with `true`. Here we are referring to nodes with `"nodeType": "door"`, which excludes sand entrances (which instead have `"nodeType": "entrance"`). These implicit strats have the same door node for both their `from` and `to` nodes.
```

Comment:
You should be able to mix the true/false of these two conditions, right? i.e. if you needed artificial morph upon entry, but then went through a door with no cap, that you couldn't get to unless you unmorphed along the way? (I don't believe any of these exist in the vanilla tho)

Reply:
Yes, that is the idea. The implicit strats don't mix them, but explicit strats could.